tutorials/2_Gazebo_and_ROS.ipynb

```diff
@@ -0,0 +1,1330 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "<!-- Differential drive controller  -->\n",
+    "  <gazebo>\n",
+    "    <plugin name=\"differential_drive_controller\" filename=\"libgazebo_ros_diff_drive.so\">\n",
+    "\n",
+    "      <rosDebugLevel>Debug</rosDebugLevel>\n",
+    "      <publishWheelTF>false</publishWheelTF>\n",
+    "      <robotNamespace>/</robotNamespace>\n",
+    "      <publishTf>1</publishTf>\n",
+    "      <publishWheelJointState>false</publishWheelJointState>\n",
+    "      <alwaysOn>true</alwaysOn>\n",
+    "      <updateRate>100.0</updateRate>\n",
+    "      <leftJoint>front_left_wheel_joint</leftJoint>\n",
+    "      <rightJoint>front_right_wheel_joint</rightJoint>\n",
+    "      <wheelSeparation>${2*base_radius}</wheelSeparation>\n",
+    "      <wheelDiameter>${2*wheel_radius}</wheelDiameter>\n",
+    "      <broadcastTF>1</broadcastTF>\n",
+    "      <wheelTorque>30</wheelTorque>\n",
+    "      <wheelAcceleration>1.8</wheelAcceleration>\n",
+    "      <commandTopic>cmd_vel</commandTopic>\n",
+    "      <odometryFrame>odom</odometryFrame> \n",
+    "      <odometryTopic>odom</odometryTopic> \n",
+    "      <robotBaseFrame>base_footprint</robotBaseFrame>\n",
+    "\n",
+    "\n",
+    "    </plugin>\n",
+    "  </gazebo> "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<p>With this plugin, we will be able to control the robot.</p>"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<p>Now we need to create add our robot model in the launch file we created earlier to spawn the model. Add below in the diff_wheeled_gazebo.launch file </p>\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "<!-- urdf xml robot description loaded on the Parameter Server-->\n",
+    "  <param name=\"robot_description\" command=\"$(find xacro)/xacro.py '$(find diff_wheeled_robot_description)/urdf/diff_wheeled_robot.xacro'\" /> \n",
+    "\n",
+    "\n",
+    "  <!-- Run a python script to the send a service call to gazebo_ros to spawn a URDF robot -->\n",
+    "  <node name=\"urdf_spawner\" pkg=\"gazebo_ros\" type=\"spawn_model\" respawn=\"false\" output=\"screen\"\n",
+    "\targs=\"-urdf -model diff_wheeled_robot -param robot_description\"/> "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<p>The first tag will first call the xacro script to convert of xacro description into an actual URDF. This URDF is then inserted into a ROS parameter called “robot_description” (this is a standard name used by many ROS tools).</p>\n",
```

Comment:
The first tag will call the xacro to convert the xacro description into a actual URDF.